packages/backend-data/src/resolve_entry_path.ts

```diff
@@ -1,15 +1,25 @@
-import { FilePathExtractor } from '@aws-amplify/platform-core';
+import {
+  AmplifyUserError,
+  FilePathExtractor,
+} from '@aws-amplify/platform-core';
 import { dirname, join } from 'path';
 import type { JsResolverEntry } from '@aws-amplify/data-schema-types';
+import { AmplifyDataError } from './types.js';
 
 /**
  * Resolve JS Resolver Handler or Sql Reference Handler entry path to absolute path
  * @param entry handler entry
  * @returns resolved absolute file path
  */
 export const resolveEntryPath = (entry: JsResolverEntry): string => {
-  const unresolvedImportLocationError = new Error(
-    'Could not determine import path to construct absolute code path from relative path. Consider using an absolute path instead.'
+  const unresolvedImportLocationError = new AmplifyUserError<AmplifyDataError>(
+    'UnresolvedEntryPath',
+    {
+      message:
+        'Could not determine import path to construct absolute code path from relative path: ' +
+        JSON.stringify(entry),
+      resolution: 'Consider using an absolute path instead.',
+    }
```

Comment:
We could put the entry values in the `details` part of the error. Something like:

```suggestion
    {
      message: 'Could not determine import path to construct absolute code path from relative path.',
      details: `Relative Path: ${entry.relativePath}${os.EOL}Import Line: ${entry.importLine}`,
      resolution: 'Consider using an absolute path instead.',
    }
```

Reply:
Sorry, PR was set to auto merge, otherwise I'd have applied the suggestions